From b5d5de387ed27c91682c908bf8a99937d9c01ee3 Mon Sep 17 00:00:00 2001
From: Emergi Idan <idan.emergi@intel.com>
Date: Mon, 2 Dec 2019 12:49:38 +0200
Subject: [PATCH] PONRTSYS-6251: ttl value should be taken from tunnel and not
 inner header

Stop copy TTL from inner packet to outer packet on fast path
instead, use the TTL on the egress packet as is

Signed-off-by: Emergi Idan <idan.emergi@intel.com>
---
 .../ppv4/pp-api/pp-session-mgr/pp_session_mgr.c    |  6 -----
 .../net/ppv4/pp-hal/modifier/modifier_recipes.c    | 28 ----------------------
 2 files changed, 34 deletions(-)

diff --git a/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c b/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
index 2f423a7f4325..620a4386a706 100644
--- a/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
+++ b/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
@@ -2497,12 +2497,6 @@ static void __smgr_newhdr_ipv4_update(struct sess_info *sess)
 
 	iphdr = (struct iphdr *)(si->bce.nhdr + si->dpu.nhdr_l3_off);
 
-	/* if inner ip exist, the ttl will be taken
-	 * from the inner layer by the modifier
-	 */
-	if (__smgr_is_iptun_encp_sess(sess))
-		iphdr->ttl = 0;
-
 	iphdr->id       = 0;             /* will be set by the modifier */
 	iphdr->tot_len  = 0;             /* will be set by the modifier */
 	iphdr->frag_off &= htons(IP_DF); /* preserve only the DF bit    */
diff --git a/drivers/net/ppv4/pp-hal/modifier/modifier_recipes.c b/drivers/net/ppv4/pp-hal/modifier/modifier_recipes.c
index fadbbcf3149d..2f11fda80b16 100644
--- a/drivers/net/ppv4/pp-hal/modifier/modifier_recipes.c
+++ b/drivers/net/ppv4/pp-hal/modifier/modifier_recipes.c
@@ -682,14 +682,6 @@ static s32 mod_rcp_ipv4_routed_set(s32 rcp_idx, enum mod_eu_stw_mux_sel l3_off,
 					      FIELD_SIZEOF(struct ipv6hdr,
 							   payload_len));
 
-		/* Replace L3 hop limit with inner ttl */
-		ret |= mod_reg_to_new_hdr_cpy(&rcp, tmp_ttl,
-					      MOD_PKT_OFF_L3_EXT,
-					      offsetof(struct ipv6hdr,
-						       hop_limit),
-					      FIELD_SIZEOF(struct ipv6hdr,
-							   hop_limit));
-
 	} else if (nhdr == RCP_NEW_HDR_V4) {
 		/* Update ipv4 id */
 		ret |= mod_reg_to_new_hdr_cpy(&rcp, MOD_GLB_EXT_IPV4_ID,
@@ -704,12 +696,6 @@ static s32 mod_rcp_ipv4_routed_set(s32 rcp_idx, enum mod_eu_stw_mux_sel l3_off,
 					      FIELD_SIZEOF(struct iphdr,
 							   tot_len));
 
-		/* Replace ipv4 ttl with inner hop limit  */
-		ret |= mod_reg_to_new_hdr_cpy(&rcp, tmp_ttl,
-					      MOD_PKT_OFF_L3_EXT,
-					      offsetof(struct iphdr, ttl),
-					      FIELD_SIZEOF(struct iphdr, ttl));
-
 		/* Replace L3 CSUM */
 		ret |= mod_reg_to_new_hdr_cpy(&rcp, si_nhdr_csum,
 					      MOD_PKT_OFF_L3_EXT,
@@ -917,14 +903,6 @@ static s32 mod_rcp_ipv6_routed_set(s32 rcp_idx, enum mod_eu_stw_mux_sel l3_off,
 					      FIELD_SIZEOF(struct ipv6hdr,
 							   payload_len));
 
-		/* Replace L3 hop limit with inner ttl */
-		ret |= mod_reg_to_new_hdr_cpy(&rcp, tmp_ttl,
-					      MOD_PKT_OFF_L3_EXT,
-					      offsetof(struct ipv6hdr,
-						       hop_limit),
-					      FIELD_SIZEOF(struct ipv6hdr,
-							   hop_limit));
-
 	} else if (nhdr == RCP_NEW_HDR_V4) {
 		/* Update ipv4 id */
 		ret |= mod_reg_to_new_hdr_cpy(&rcp, MOD_GLB_EXT_IPV4_ID,
@@ -939,12 +917,6 @@ static s32 mod_rcp_ipv6_routed_set(s32 rcp_idx, enum mod_eu_stw_mux_sel l3_off,
 					      FIELD_SIZEOF(struct iphdr,
 							   tot_len));
 
-		/* Replace ipv4 ttl with inner hop limit  */
-		ret |= mod_reg_to_new_hdr_cpy(&rcp, tmp_ttl,
-					      MOD_PKT_OFF_L3_EXT,
-					      offsetof(struct iphdr, ttl),
-					      FIELD_SIZEOF(struct iphdr, ttl));
-
 		/* Replace L3 CSUM */
 		ret |= mod_reg_to_new_hdr_cpy(&rcp, si_nhdr_csum,
 					      MOD_PKT_OFF_L3_EXT,
